ml-pipeline/ml-pipeline/pipeline-apiserver.libsonnet

```diff
@@ -205,66 +202,9 @@
       },
       rules: [
         {
-          apiGroups: [""],
-          resources: [
-            "pods",
-            "pods/exec",
-          ],
-          verbs: [
-            "create",
-            "get",
-            "list",
-            "watch",
-            "update",
-            "patch",
-          ],
-        },
-        {
-          apiGroups: [""],
-          resources: [
-            "secrets",
-          ],
-          verbs: [
-            "get",
-          ],
-        },
-        {
-          apiGroups: [""],
-          resources: [
-            "configmaps",
-          ],
-          verbs: [
-            "get",
-            "watch",
-            "list",
-          ],
-        },
-        {
-          apiGroups: [
-            "",
-          ],
-          resources: [
-            "persistentvolumeclaims",
-          ],
-          verbs: [
-            "create",
-            "delete",
-          ],
-        },
-        {
-          apiGroups: [
-            "argoproj.io",
-          ],
-          resources: [
-            "workflows",
-          ],
-          verbs: [
-            "get",
-            "list",
-            "watch",
-            "update",
-            "patch",
-          ],
+          apiGroups: ["*"],
+          resources: ["*"],
+          verbs: ["*"],
```

Comment:
Does this mean the pipeline can also change/delete anything already deployed in the cluster?

Reply:
yes. FYI this is what's happening before we switch to kf namespace, when using default namespace. the default service account have cluster admin permission. Since we don't have multi-tenancy and authorization story, this is ok. when we need to support authorization, we should use namespace as security boundary.

Reply:
This looks really dangerous while our containers are everything but hermetic.

We should not let containers just directly create K8s objects and launch CRDs. I think this was discussed during one of the design meetings. We should put this genie back into the bottle ASAP.

I'm sorry that this blocks some scenarios short term, but this is important in long term.

Can we have a backend endpoint for launching CRDs? To launch a CRD, the Pod needs to at least identify itself to the backend so that we can verify it.

Reply:
what credential can backend API use for authorization? 
I think the authorization model worth a discussion but it should be a product wise effort.

FYI here is jupyter role's scope. 
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/jupyter/jupyter.libsonnet#L252

(I think the range here is too broad. I'll use the same scope as jupyter one)